### PR TITLE
[C] Declare a volatile pointer to `aeron_image_list_stct` instead of a pointer to a volatile struct.

### PR DESCRIPTION
**Pull request by vyazelenko**

### CODE REVIEW

File: aeron-client/src/main/c/aeron_subscription.c [hunk 1]
```diff
@@ -119,7 +119,7 @@ int aeron_subscription_close(
     return 0;
 }
 
-int aeron_subscription_alloc_image_list(volatile aeron_image_list_t **image_list, size_t length)
+int aeron_subscription_alloc_image_list(aeron_image_list_t * volatile * image_list, size_t length)
```
Comment (mikeb01) [inline aeron-client/src/main/c/aeron_subscription.c:122]: With our coding style should it be:
```
int aeron_subscription_alloc_image_list(aeron_image_list_t *volatile *image_list, size_t length)
```

**Comment by mikeb01** [comment]
With in a number of methods in the `aeron_subscription.c` file (e.g. `aeron_subscription_controlled_poll`) we are declaring the variable:
```
volatile aeron_image_list_t *image_list;
```
Should that be
```
aeron_image_list_t *volatile image_list;
```

**Comment by vyazelenko** [comment]
@mikeb01 I've addressed all of the issues and replace all volatile struct occurrences.

**Comment by mikeb01** [comment]
I'll have a look again today.

**Pull request merged**